menas/src/main/scala/za/co/absa/enceladus/menas/utils/converters/model0/Serializer.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.menas.utils.converters.model0
+
+import java.io.ByteArrayOutputStream
+
+import com.fasterxml.jackson.databind.{ObjectMapper, SerializationFeature}
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import org.apache.spark.sql.catalyst.parser.CatalystSqlParser
+import org.apache.spark.sql.types._
+import org.json4s.{Formats, NoTypeHints}
+import org.json4s.jackson.Serialization
+
+/**
+  * This is the object for deserializing Model 0 version of Enceladus Schema and convert it to Spark StructType
+  */
+object Serializer {
+  private val objectMapper = new ObjectMapper()
+    .registerModule(DefaultScalaModule)
+    .registerModule(new JavaTimeModule())
+    .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
+
+  implicit private val formatsJson: Formats = Serialization.formats(NoTypeHints).withBigDecimal
+
+  /**
+    * Converts a JSON in Model 0 format to a Spark StructType
+    */
+  def convertFromModel0ToStructType(json: String): StructType = convertToStructType(deserialize(json))
```

Comment:
I have a feeling `Model0` might be a bit too much concrete detail. It tells me that this serializer might also handle conversion from `Model1`, `Model2`, etc, but why is it in package `model0` then? If the package isn't explicit enough, I think this information should be made clear in the class name. I like how you use `model0.Serializer` where this is being called, so I don't think even that is necessary.

This method seems to introduce a noisy wrapper for `convertToStructType`. I would rename it to overload `convertToStructType(json: String)`, thus giving options if you pass in the Schema or the json directly.